python/hail/expr/functions.py

```diff
@@ -640,6 +666,25 @@ def parse_locus(s, reference_genome=None):
     return construct_expr(ApplyMethod('Locus({})'.format(reference_genome.name), s._ast), TLocus(reference_genome),
                           s._indices, s._aggregations, s._joins, s._refs)
 
+@typecheck(pl=expr_list)
+def pl_dosage(pl):
+    """
+    Return expected genotype dosage from array of Phred-scaled genotype
+    likelihoods with uniform prior. Only defined for bi-allelic variants. The
+    `pl` argument must be length 3.
+
+    Parameters
+    ----------
+    pl : :obj:`.ArrayNumericExpression`
+        Length 3 list of bi-allelic Phred-scaled genotype likelihoods
+
+    Returns
+    -------
+    :class:`.Float64Expression`
+    """
+    if not isinstance(pl, ArrayNumericExpression):
+        raise TypeError("Function 'pl_dosage' expects an 'ArrayNumericExpression'. Found {}.".format(type(pl)))
```

Comment:
@tpoterba How is this? I doesn't seem to fit in with other type errors I've seen, but I couldn't think of any other way to do it.

Reply:
I think using the Expression class names is going to be very confusing for the user. How about:
```python
element_type = pl.dtype.element_type
if not is_numeric(element_type):
    raise TypeError("function 'pl_disage' expects an expression of type 'Array[numeric]', found '{}'".format(pl.dtype))
```

Reply:
not quite sure about the 'Array[numeric]' but the rest seems good